### ISSUE TITLE
Upstream PacketStream to Clash.Protocols

### ISSUE THREAD

**Issue by rowanG077** [GiPHouse/qbaylogic-clash-based-macipudp-stack-spring24#122]
**Feature description**
Beside the ethernet stack qbaylogic would really like to have the `PacketStream` protocol inside Clash.Protocols. This issue is to upstream the protocol + all our generic modules. 

**Expected behaviour**
PR to Clash.Protocols is made and accepted.  

**Expected steps**
Clone the clash protocols repo and move our PacketStream protocol to it and make a PR.

**Additional Context**

**Comment by MatthijsMu** [comment]
I will get the code working, retain the current file structure and see in the client's review what components they would like to see merged into one file. I can imagine the up+downconverter and de+packetizer into one (perhaps also with Df), but it may be useless to do this prematurely.

**Comment by rowanG077** [comment]
I would not merge everything into Df, definitely a separate module. Theoretically we could merge it into one file but personally I would like to keep at least a bit split up. Mainly because a single HS module is compiled sequentially whereas multiple HS files can be compiled in parallel.

I would first wait on what @JLaumen comes back with, he had a good proposal in structure he is implementing now.

**Comment by rowanG077** [comment]
Also note that this issue is currently not in the sprint. So the issues that are open which are in the sprint have higher priority.

**Comment by lmbollen** [comment]
> I can imagine the up+downconverter, de+packetizer want to be merged into one file (perhaps also with Df), but it may be useless to do this prematurely. @lmbollen could you tell me more about the filestructure that you would like to see?

When there's a pull request, we'll give feedback. Don't worry too much about it for now :)

**Comment by JLaumen** [comment]
@MatthijsMu The relevant modules have been moved to `Protocols.Extra.Packetstream`, see  #139  (branch of issue #115 )

**Comment by MatthijsMu** [comment]
oh, that's unfortunate I already did work on this yesterday. Should it remain in Protocols.Extra.Packetstream or should it become Protocols.Packetstream on clash-protocols?

**Comment by rowanG077** [comment]
It should become `Protocols.Packetstream`. The "extra" pattern is done so there will be no namespace clashes with the original. So once it's upstreamed it's no longer extra. But like I said I only want to pick this up once we run out of issues in the sprint.

**Comment by MatthijsMu** [comment]
okay I'll continue with the ip to eth transformer for now
